.github/workflows/pull-request.yml

```diff
@@ -0,0 +1,273 @@
+name: "pull-request"
+
+# https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request
+on:
+  pull_request:
+    types:
+      - opened
+    branches:
+      - /^master(-patch.*)?$/
+      - /^develop(-patch.*)?$/
+      - master
+      - develop
+
+env:
+  AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+  AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+  TRAVIS_BRANCH: ${{ github.ref_name }}
+  GITHUB_BRANCH: ${{ github.ref_name }}
+  TRAVIS_BUILD_DIR: ${{ github.workspace }}
+  TRAVIS_COMMIT: ${{ github.sha }}
+  BUILD_ID: ${{ github.run_id }}
+  TRAVIS_RUN_NUMBER: ${{ github.run_attempt }}
+  TRAVIS_BUILD_NUMBER: ${{ github.run_id }}
+  JOB_ID: ${GITHUB_RUN_ID}
+  PROXY_HOST_BPM: ${{ secrets.E2E_HOST }}
+  E2E_HOST_APA: ${{ secrets.E2E_HOST_APA }}
+  E2E_HOST: ${{ secrets.E2E_HOST }}
+  E2E_USERNAME: ${{ secrets.E2E_ADMIN_EMAIL_IDENTITY }}
+  E2E_PASSWORD: ${{ secrets.E2E_PASSWORD }}
+  E2E_ADMIN_EMAIL_IDENTITY: ${{ secrets.E2E_ADMIN_EMAIL_IDENTITY }}
+  E2E_ADMIN_PASSWORD_IDENTITY: ${{ secrets.E2E_ADMIN_PASSWORD_IDENTITY }}
+  USERNAME_ADF: ${{ secrets.E2E_USERNAME }}
+  PASSWORD_ADF: ${{ secrets.E2E_PASSWORD }}
+  URL_HOST_ADF: ${{ secrets.URL_HOST_ADF }}
+  IDENTITY_ADMIN_EMAIL: ${{ secrets.E2E_ADMIN_EMAIL_IDENTITY }}
+  IDENTITY_ADMIN_PASSWORD: ${{ secrets.E2E_ADMIN_PASSWORD_IDENTITY }}
+  AWS_S3_BUCKET_ACTIVITI_LICENSE: ${{ secrets.AWS_S3_BUCKET_ACTIVITI_LICENSE }}
+  HOST_SSO: ${{ secrets.HOST_SSO }}
+  LOG_LEVEL: "ERROR"
+  E2E_LOG_LEVEL: "ERROR"
+  E2E_MODELER_USERNAME: ${{ secrets.E2E_MODELER_USERNAME }}
+  E2E_MODELER_PASSWORD: ${{ secrets.E2E_MODELER_PASSWORD }}
+  EXTERNAL_ACS_HOST: ${{ secrets.EXTERNAL_ACS_HOST }}
+  E2E_DEVOPS_USERNAME: ${{ secrets.E2E_DEVOPS_USERNAME }}
+  E2E_DEVOPS_PASSWORD: ${{ secrets.E2E_DEVOPS_PASSWORD }}
+  USERNAME_SUPER_ADMIN_ADF: ${{ secrets.USERNAME_SUPER_ADMIN_ADF }}
+  PASSWORD_SUPER_ADMIN_ADF: ${{ secrets.PASSWORD_SUPER_ADMIN_ADF }}
+  HR_USER: ${{ secrets.HR_USER }}
+  HR_USER_PASSWORD: ${{ secrets.HR_USER_PASSWORD }}
+
+jobs:
+  main:
+    # long timeout required when cache has to be recreated
+    timeout-minutes: 30
+    name: "Build Components"
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0 # Fetch all history for all tags and branches
+      - uses: ./.github/actions/setup
+      - run: echo $TRAVIS_EVENT_TYPE
+      - run: echo $S3_NODE_MODULES_CACHE_ID
+      - run: echo $HEAD_COMMIT_HASH
+      - run: echo $NX_CALCULATION_FLAGS
+      - run: echo NODE_OPTIONS="--max-old-space-size=5120" >> $GITHUB_ENV
+      - run: npm ci
```

Comment:
the npm ci needs to be called only once as part of the setup, all the other jobs needs to rely on the cache!